openstack_plugin/resources/network/router.py

```diff
@@ -115,11 +115,13 @@ def _handle_external_router_resource(openstack_resource):
     is connected to external network node resource
     :param openstack_resource: Instance of openstack router resource
     """
-    remote_router = openstack_resource.get()
+    network_resource = \
+        OpenstackNetwork(client_config=openstack_resource.client_config,
+                         logger=ctx.logger)
     rel_network_id = \
-        _get_connected_external_network_from_relationship(openstack_resource)
+        _get_connected_external_network_from_relationship(network_resource)
     ext_network_id = \
-        _get_external_network_id(remote_router.external_gateway_info)
+        _get_external_network_id(network_resource.external_gateway_info)
```

Comment:
We still need the `remote_router` since we need to get the value for `external_gateway_info` which at the end of the day get the external network id and then compare it with external network from the realtionship